falco.yaml

```
@@ -290,6 +291,34 @@ outputs:
   rate: 0
   max_burst: 1000
 
+# [Experimental] `queue_capacity_outputs`
+#
+# Falco utilizes tbb::concurrent_bounded_queue for handling outputs, and this parameter
+# allows you to customize the queue capacity. Please refer to the official documentation:
+# https://oneapi-src.github.io/oneTBB/main/tbb_userguide/Concurrent_Queue_Classes.html.
+# On a healthy system with optimized Falco rules, the queue should not fill up.
+# If it does, it is most likely happening due to the entire event flow being too slow,
+# indicating that the server is under heavy load.
+# 
+# Lowering the number of items can prevent memory from steadily increasing until the OOM
+# killer stops the Falco process. We provide recovery actions to self-limit or self-kill
+# in order to handle this situation earlier, similar to how we expose the kernel buffer size
+# as a parameter.
+# However, it will not address the root cause of the event pipe not keeping up.
+# 
+# `items`: the maximum number of items allowed in the queue, defaulting to 0. This means that
+# the queue is unbounded.
+# You can experiment with values greater or smaller than the anchor value 1000000.
+# 
+# `recovery`: the strategy to follow when the queue becomes filled up. This also applies when
+# the queue is unbounded, and all available memory on the system is consumed.
+# recovery: 0 means continue.
+# recovery: 1 means simply exit (default behavior).
+# recovery: 2 means empty the queue and then continue.
+queue_capacity_outputs:
+  items: 0
```

```suggestion
outputs_queue:
  capacity: 0
```

I'd prefer this way just because it's more consistent with other options (I know - we are not consistent in all cases), and it can handle future queue options.

Alternatively, we can put it under `outputs` (since the purpose is similar).
```
outputs:
  rate: 0
  max_burst: 1000
  queue_capacity: 0
```

---

We can put it under outputs, where to put the recovery then and what wording?

---

> We can put it under outputs, where to put the recovery then and what wording?

Sorry, I did not notice that. 
:thinking: At this point, I believe this :point_down: would be preferable

```
outputs_queue:
	  capacity: 0
	  recovery: exit
```	  

PS
As we do with `log_level` , I'd prefer strings for enums, not just numbers
https://github.com/falcosecurity/falco/blob/e0c6c9de8463f2ab8fc1de50cfec8710e646b439/falco.yaml#L488

---

Ok will do!